Dockerfile

```diff
@@ -0,0 +1,82 @@
+# This dockerfile expects proxies to be set via --build-arg if needed
```

Comment:
we have a .docker folder:
https://github.com/ros-planning/moveit2/tree/master/.docker

@mlautman has been setting up docker for ROS2:
https://github.com/ros-planning/moveit2/pull/46

Reply:
Optimizations of the CI config heavily rely upon the assumed setup of this Dockerfile, so they go hand-in-hand. Perhaps @mlautman could mention what changes should be added into this CI dockerfile?

Reply:
I would recommend testing with the [.docker/ci/Dockerfile](https://github.com/ros-planning/moveit2/blob/90d331e536b9278c3d2c451cd3d1b6ca531f73db/.docker/ci/Dockerfile) in the [MoveIt2 Ci PR](https://github.com/ros-planning/moveit2/pull/46) as it is very close to being merged in.

Reply:
I made some comments on https://github.com/ros-planning/moveit2/pull/46#pullrequestreview-216485284 that the Dockerfile here also addresses. However, like I mentioned, the Dockerfile here and config go rather hand-in-hand due to the build caching strategy optimizations. That said, I'm not opposed to multiple dockerfiles; we could have one for the existing travis setup and one for CircleCI.

Reply:
We certainly can have multiple (we currently have about 4) but it is important that we remove redundancies wherever possible. Looking through the Dockerfile (albeit rather briefly) it seems that the docker container created isn't very different from [moveit/moveit2:crystal-source](https://github.com/ros-planning/moveit2/blob/moveit2-ci/.docker/source/Dockerfile)
The main difference being that this Dockerfile builds the workspace whiel crystal-source does not.  (Although it certainly could)

Reply:
@davetcoleman and myself were just discussing this and it makes sense, at least in the short term, to have multiple dockerfiles with one for the existing travis setup and one for CricleCi

Reply:
Agreed, I would also like to prune Dockerfile redundancies. I have an idea to address this later using build args in the FROM calls and overriding the Dockerhub build hooks. See bfdbc03 and https://docs.docker.com/docker-hub/builds/advanced as I linked to above with https://github.com/ruffsl/navigation2/tree/foo/.dockerhub/devel

Reply:
Can we move this file to our ``.docker`` folder at least?

Reply:
```suggestion
# Docker specifically for CircleCi setup, not for Travis
#
# This dockerfile expects proxies to be set via --build-arg if needed
```

Reply:
> Can we move this file to our .docker folder at least?

The Dockerfile is kept in the root of the project so that the build context may encompass the entire repo so that we can COPY all the separate source files into the image. It is possible to have different Dockerfile and build context locations, but from a user perspective, the expected build context location is no longer obvious. DockerHub has recently revamped the build rules setting, allowing us to specify separate paths, and we can document the use required `pwd` and `--file` for uses building locally.

https://docs.docker.com/engine/reference/commandline/build/#extended-description
https://docs.docker.com/engine/reference/commandline/build/#text-files